packages/material-ui/src/Modal/TrapFocus.js

```diff
@@ -0,0 +1,146 @@
+/* eslint-disable consistent-return, jsx-a11y/no-noninteractive-tabindex */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import warning from 'warning';
+import RootRef from '../RootRef';
+import ownerDocument from '../utils/ownerDocument';
+
+function TrapFocus(props) {
+  const { disableEnforceFocus, disableAutoFocus, disableRestoreFocus, isEnabled, open } = props;
+  const rootRef = React.useRef();
+  const ignoreNextEnforceFocus = React.useRef();
+  const sentinelStart = React.useRef();
+  const sentinelEnd = React.useRef();
+  const lastFocus = React.useRef();
+
+  React.useEffect(() => {
+    if (!open) {
+      return;
+    }
+
+    const doc = ownerDocument(rootRef.current);
+    const currentActiveElement = doc.activeElement;
```

Comment:
Hi,

I'm already tested it and noticed some issue with it.
If I used `autoFocus` on some input in content `doc.activeElement` return that input, not the previous one.
Or I do something wrong :)

Can you check this case ?

Regards,
Piotr S

Reply:
Thanks for the report, I'm looking into ti.

Reply:
Solved.

Reply:
#14661